# Limpieza de datos
Crearemos los archivos necesarios para poder realizar los análisis correspondientes.

In [1]:
import pandas as pd

## Consumo de Productos Derivados del Petróleo

En la fila 7 se encuentran los títulos de las columnas. Con datos hasta la fila 300.

En la columna A se encuentra la fecha, en la columna L se encuentra la gasolina superior, en la K la gasolina regular, en la G el diesel bajo azufre y el gas propano en la I.

In [2]:
df = pd.read_excel('data/raw/CONSUMO.xlsx', header=6, nrows=294, usecols='A,G,I,K,L')

In [3]:
df.tail()

Fecha  Diesel bajo azufre  Gas licuado de petróleo  Gasolina regular  \
288 2024-01-01          1359012.49                548124.45         830708.13   
289 2024-02-01          1340174.42                526897.85         818740.16   
290 2024-03-01          1393324.52                523990.91         870771.70   
291 2024-04-01          1428143.44                531880.19         847353.15   
292 2024-05-01          1401052.37                536754.38         894533.14   

     Gasolina superior  
288          658083.66  
289          654059.60  
290          671997.05  
291          633520.57  
292          692427.94

Renombraremos las columnas para facilitar su manipulación.

In [4]:
df.columns = ['Fecha', 'Diesel', 'Propano', 'Regular', 'Super']
df.head()

Fecha  Diesel        Propano    Regular      Super
0 2000-01-01     0.0  194410.476190  202645.20  308156.82
1 2000-02-01     0.0  174710.552381  205530.96  307766.31
2 2000-03-01     0.0  189234.066667  229499.56  331910.29
3 2000-04-01     0.0  174330.607143  210680.40  315648.08
4 2000-05-01     0.0  191745.147619  208164.34  319667.97

Convertimos a una serie de tiempo

In [5]:
df['Fecha'] = pd.to_datetime(df['Fecha'])
df.set_index('Fecha', inplace=True)
df.sort_index(inplace=True)

In [6]:
df.tail()

Diesel    Propano    Regular      Super
Fecha                                                  
2024-01-01  1359012.49  548124.45  830708.13  658083.66
2024-02-01  1340174.42  526897.85  818740.16  654059.60
2024-03-01  1393324.52  523990.91  870771.70  671997.05
2024-04-01  1428143.44  531880.19  847353.15  633520.57
2024-05-01  1401052.37  536754.38  894533.14  692427.94

## Importación de Productos Derivados del Petróleo

En la fila 7 se encuentran los títulos de las columnas. Con datos hasta la fila 288.

En la columna A se encuentra la fecha, en la columna L se encuentra la gasolina superior, en la K la gasolina regular, en la G el diesel bajo azufre y el gas propano en la I.

In [7]:
df = pd.read_excel('data/raw/IMPORTACION.xlsx', header=6, nrows=281, usecols='A,G,I,K,L')

In [8]:
df.tail()

Fecha  Diesel bajo azufre  Gas licuado de petróleo  Gasolina regular  \
276 2024-01-01          1409097.15                701570.80         914133.32   
277 2024-02-01          1236861.75                916541.70         740662.25   
278 2024-03-01          1477038.00                675157.48         838270.93   
279 2024-04-01          1294706.12                473940.66         886132.77   
280 2024-05-01          1470870.09                684864.46         939656.18   

     Gasolina superior  
276          712333.33  
277          650360.11  
278          620077.74  
279          687017.96  
280          696970.30

Renombraremos las columnas para facilitar su manipulación.

In [9]:
df.columns = ['Fecha', 'Diesel', 'Propano', 'Regular', 'Super']
df.head()

Fecha  Diesel        Propano    Regular      Super
0 2001-01-01     0.0  194065.738095  177776.50  373963.96
1 2001-02-01     0.0  170703.380952  123115.99  243091.07
2 2001-03-01     0.0  161837.371429  161726.42  312084.38
3 2001-04-01     0.0  163048.642857  127338.74  285054.89
4 2001-05-01     0.0  171518.861905  168730.19  300913.67

Convertimos a una serie de tiempo

In [10]:
df['Fecha'] = pd.to_datetime(df['Fecha'])
df.set_index('Fecha', inplace=True)
df.sort_index(inplace=True)

In [11]:
df.tail()

Diesel    Propano    Regular      Super
Fecha                                                  
2024-01-01  1409097.15  701570.80  914133.32  712333.33
2024-02-01  1236861.75  916541.70  740662.25  650360.11
2024-03-01  1477038.00  675157.48  838270.93  620077.74
2024-04-01  1294706.12  473940.66  886132.77  687017.96
2024-05-01  1470870.09  684864.46  939656.18  696970.30

## Precios de los Productos Derivados del Petróleo

Un formato no muy amigable, se leerá cada producto por separado. Luego, se unirán en un solo DataFrame.

### Gasolina Superior

In [23]:
df_super = pd.read_excel('data/raw/PRECIOs.xlsx', header=22, nrows=12, usecols='A:X')

In [24]:
df_super.tail()

MES/AÑO   2002   2003   2004   2005   2006   2007   2008   2009   2010  \
7       AGOSTO  15.29  15.14  18.71  22.61  27.86  27.52  35.41  26.16  27.69   
8   SEPTIEMBRE  15.25  16.55  18.60  25.21  26.97  26.78  34.45  26.28  26.93   
9      OCTUBRE  15.63  16.27  19.20  26.85  24.45  27.02  30.77  25.57  28.50   
10   NOVIEMBRE  15.76  16.00  20.04  25.50  22.90  27.99  25.76  27.33  28.47   
11   DICIEMBRE  15.49  15.49  19.70  22.93  22.88  28.73  22.07  26.93  29.43   

    ...   2015   2016   2017   2018   2019   2020   2021   2022   2023  2024  
7   ...  24.37  21.54  22.84  27.06  25.16  21.21  28.91  35.28  37.34   NaN  
8   ...  22.42  22.17  24.42  27.31  24.98  21.08  29.19  34.24  37.10   NaN  
9   ...  22.00  22.54  23.55  27.66  24.72  21.15  29.99  36.35  35.58   NaN  
10  ...  21.12  21.85  24.48  24.95  24.66  20.11  30.48  34.90  31.03   NaN  
11  ...  20.18  22.80  24.22  22.93  24.54  21.61  28.77  33.63  29.55   NaN  

[5 rows x 24 columns]

In [25]:
df_regular = pd.read_excel('data/raw/PRECIOs.xlsx', header=43, nrows=12, usecols='A:X')

In [26]:
df_regular.tail()

MES/AÑO   2002   2003   2004   2005   2006   2007   2008   2009   2010  \
7       AGOSTO  14.88  14.59  18.22  22.09  27.35  26.82  34.74  25.17  26.66   
8   SEPTIEMBRE  14.83  16.05  18.09  24.66  26.46  25.80  33.78  25.30  25.91   
9      OCTUBRE  15.22  15.80  18.75  26.33  23.95  26.05  30.14  24.62  27.49   
10   NOVIEMBRE  15.34  15.52  19.55  24.99  22.43  27.02  24.79  26.34  27.48   
11   DICIEMBRE  15.06  15.06  19.20  22.42  22.40  27.97  21.30  25.94  28.43   

    ...   2015   2016   2017   2018   2019   2020   2021   2022   2023  2024  
7   ...  22.88  20.05  21.55  25.92  23.97  20.40  28.11  34.19  35.84   NaN  
8   ...  20.94  20.67  23.13  26.11  23.79  20.28  28.49  33.33  35.60   NaN  
9   ...  20.51  21.05  22.26  26.48  23.61  20.35  29.29  35.44  34.07   NaN  
10  ...  19.64  20.35  23.25  23.75  23.57  19.31  29.78  34.90  29.54   NaN  
11  ...  18.69  21.30  23.20  21.74  23.54  20.81  28.03  32.06  28.06   NaN  

[5 rows x 24 columns]

In [27]:
df_diesel = pd.read_excel('data/raw/PRECIOs.xlsx', sheet_name=1, header=1, nrows=12, usecols='A:X')

In [28]:
df_diesel.tail()

MES/AÑO   2002   2003   2004   2005   2006   2007   2008   2009   2010  \
7       AGOSTO   9.85  10.25  14.88  17.63  21.43  21.61  34.98  21.41  23.17   
8   SEPTIEMBRE   9.81  10.92  14.83  19.01  20.99  21.24  32.47  21.46  22.60   
9      OCTUBRE  10.07  10.56  15.22  20.27  19.73  22.77  28.88  20.97  24.55   
10   NOVIEMBRE  10.14  10.72  15.34  20.85  18.76  25.00  24.72  23.13  25.17   
11   DICIEMBRE  10.06  11.00  15.06  20.44  18.63  25.93  20.71  23.05  26.65   

    ...   2015   2016   2017   2018   2019   2020   2021   2022   2023  2024  
7   ...  17.40  16.22  17.55  22.06  20.21  16.62  22.91  33.89  31.85   NaN  
8   ...  17.63  16.88  18.30  22.98  20.99  15.45  23.56  30.70  33.46   NaN  
9   ...  17.65  17.57  19.10  24.19  21.20  15.01  25.46  38.33  33.72   NaN  
10  ...  17.21  17.18  19.93  22.87  20.93  15.07  26.02  38.89  31.12   NaN  
11  ...  15.41  18.15  20.07  20.72  21.09  17.23  24.36  34.52  29.29   NaN  

[5 rows x 24 columns]